internal/clients/http/sources/sources_client.go

```diff
@@ -83,7 +74,7 @@ func (c *sourcesClient) Ready(ctx context.Context) error {
 	return nil
 }
 
-func (c *sourcesClient) ListProvisioningSources(ctx context.Context) (*[]clients.Source, error) {
+func (c *sourcesClient) ListProvisioningSources(ctx context.Context) ([]*clients.Source, error) {
```

Comment:
This was an overlook, it should be slice of pointers, not pointer to slice. It is actually required to be able to refactor copy loop in this function effectively.